cli/cmds.go

```diff
@@ -215,6 +217,66 @@ func Make() *cobra.Command {
 	blocksCmd.Flags().BoolP("zero-terminated", "z", false, "outputs blocks separated by null separator")
 	blocksCmd.Flags().BoolP("json", "j", false, "outputs blocks in JSON format")
 
+	// options
+	grepCmd := &cobra.Command{
+		Use:   "grep [path] -- [hclgrep option]",
```

Comment:
this is not grep, grep would be a simple grep of the block ie `terrafmt grep "sku.*=.*"Basic"` or even `terrafmt grep -rt azurerm_resource_group -rn exampleRG "sku.*=.*"Basic"`

you've added your more complicated tool which is not grep, so:
```suggestion
		Use:   "hclgrep [path] -- [hclgrep option]",
```

also why is the -- needed?

Reply:
Since there are a couple of options used in `hclgrep`, which is a nontrivial work to re-implement them one by one here, not mention it might be overlapping with the existing options. As `--` is a wellknown delimiter used in Bash for indicating the end of options and the `spf13/pflag` has good support for this: [ArgsLenAt()](https://pkg.go.dev/github.com/spf13/pflag#FlagSet.ArgsLenAtDash). Using it here so that we can grab all the `hclgrep` options as a whole:

https://github.com/katbyte/terrafmt/blob/052ee4e7a9d66d04ebce5593e4cb3cdf9ce066e6/cli/cmds.go#L237-L240

Then use the hclgrep function to do the parsing:

https://github.com/katbyte/terrafmt/blob/052ee4e7a9d66d04ebce5593e4cb3cdf9ce066e6/cli/cmds.go#L251